apps/src/applab/applab.js

```diff
@@ -983,6 +984,10 @@ Applab.reset = function (first) {
  */
 function runButtonClickWrapper(callback) {
   $(window).trigger('run_button_pressed');
+
+  // Change back to default screen before we serialize. This also makes it so
+  // that we'll be back on the default screen after a reset.
+  studioApp.reduxStore.dispatch(changeScreen(elementUtils.getDefaultScreenId()));
```

Comment:
I think part of our problem here is that we're using our reduxStore to reflect the state of our currentScreenId while not running, but don't touch it while we are running. I'll think of whether I can rationalize this better and/or come up with something cleaner.